### ISSUE TITLE
pytorch linear parameters calculation

### ISSUE THREAD

**Issue by Shavell** [skyduy/CNN_keras#19]
Hi,

how can i calculate that parameters in pytorch branch?

[train.py](https://github.com/skyduy/CNN_keras/blob/pytorch/train.py#L22)

 self.fc1 = nn.Linear(48 * 6 * 27, 360)

what does it mean that parameters (48 * 6 * 27), 360 wheres coming?

ty

**Comment by LiuChuang1992** [comment]
360 means the data trained




------------------ 原始邮件 ------------------
发件人: "Shavell"<notifications@github.com>; 
发送时间: 2019年6月20日(星期四) 凌晨3:45
收件人: "skyduy/CNN_keras"<CNN_keras@noreply.github.com>; 
抄送: "Subscribed"<subscribed@noreply.github.com>; 
主题: [skyduy/CNN_keras] pytorch linear parameters calculation (#19)




Hi,
 
how can i calculate that parameters in pytorch branch?
 
train.py
 
self.fc1 = nn.Linear(48 * 6 * 27, 360)
 
what does it mean that parameters (48 * 6 * 27), 360 wheres coming?
 
ty
 
&mdash;

**Comment by Shavell** [comment]
`
        # size: 3 * 36 * 120
        self.conv1 = nn.Conv2d(3, 18, 5)  # 18 * 32 * 116
        self.pool1 = nn.MaxPool2d(2)  # 18 * 16 * 58
        self.conv2 = nn.Conv2d(18, 48, 5)  # 48 * 12 * 54
        self.pool2 = nn.MaxPool2d(2)  # 48 * 6 * 27
        self.drop = nn.Dropout(0.5)
        self.fc1 = nn.Linear(48 * 6 * 27, 360)
        self.fc2 = nn.Linear(360, 19 * 4)

`

can u explain this params how can be calculate?

ty

**Comment by skyduy** [comment]
> Hi,
> 
> how can i calculate that parameters in pytorch branch?
> 
> [train.py](https://github.com/skyduy/CNN_keras/blob/pytorch/train.py#L22)
> 
> self.fc1 = nn.Linear(48 * 6 * 27, 360)
> 
> what does it mean that parameters (48 * 6 * 27), 360 wheres coming?
> 
> ty

360 is the unit number in hidden layer.

**Comment by skyduy** [comment]
> `
> # size: 3 * 36 * 120
> self.conv1 = nn.Conv2d(3, 18, 5) # 18 * 32 * 116
> self.pool1 = nn.MaxPool2d(2) # 18 * 16 * 58
> self.conv2 = nn.Conv2d(18, 48, 5) # 48 * 12 * 54
> self.pool2 = nn.MaxPool2d(2) # 48 * 6 * 27
> self.drop = nn.Dropout(0.5)
> self.fc1 = nn.Linear(48 * 6 * 27, 360)
> self.fc2 = nn.Linear(360, 19 * 4)
> 
> `
> 
> can u explain this params how can be calculate?
> 
> ty

See the last part of [README](https://github.com/skyduy/CNN_keras/tree/pytorch#model-graph) for detail.

**Comment by Shavell** [comment]
i did but

 Conv2d(18, 48, 5) # 48 * 12 * 54

if i look last part of readme 48 * 12 * 54 how can equal to 18, 48, 5 for conv2d i don't know
